pkg/cri/server/sandbox_service.go

```diff
@@ -0,0 +1,47 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package server
+
+import (
+	"fmt"
+
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
+
+	"github.com/containerd/containerd"
+	criconfig "github.com/containerd/containerd/pkg/cri/config"
+	"github.com/containerd/containerd/sandbox"
+)
+
+type criSandboxService struct {
+	client *containerd.Client
+	config *criconfig.Config
+}
+
+func newCriSandboxService(config *criconfig.Config, client *containerd.Client) *criSandboxService {
+	return &criSandboxService{
+		client: client,
+		config: config,
+	}
+}
+
+func (c *criSandboxService) SandboxController(config *runtime.PodSandboxConfig, runtimeHandler string) (sandbox.Controller, error) {
+	ociRuntime, err := c.config.GetSandboxRuntime(config, runtimeHandler)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sandbox runtime: %w", err)
+	}
+	return c.client.SandboxController(ociRuntime.Sandboxer), nil
```

Comment:
REF: https://github.com/containerd/containerd/pull/8268/files#r1219996770

I think you should add the `SandboxMode` back and mark it as deprecated and then update `RELEASE.md`.
We should delete the `SandboxMode` in v2.1

Reply:
Or could this be a migration now

Reply:
I think it is a good idea. @dmcgowan  I have defined a ConfigMigration in cri plugin in my local codes, I can add a migration from `SandboxMode` to `Sandboxer` in that commit, and submit it. Is it ok? @fuweid